app/controllers/task_managements_controller.rb

```diff
@@ -23,6 +54,20 @@ def update
     redirect_to dashboard_path
   end
 
+  def send_email_notifications(task)
+    notif_taskee = User.find_by_id(task.taskee_id)
+    notif_tasker = User.find_by_id(task.tasker_id)
+    task_category = Task.find_by_id(task.task_id)
+    @notif = current_user.enable_notifications
+    NotificationMailer.send_notifications(
+      current_user,
+      task,
+      task_category,
+      notif_tasker,
+      notif_taskee
+    ).deliver_now if @notif == true
```

Comment:
When you have situations like this `if @notif == true`, it will make sense to return early instead of at the very end. Reason been that you don't what to speed the time and resources finding taskee, tasker and task_category before discovering that a user didn't enable notifications. You can have something like this.
```ruby
def send_email_notifications(task)
  return unless current_user.enable_notifications
  # do your things
end
```
I also noticed that you are passing a lot of arguments to `NotificationMailer.send_notifications`. This probably be a good time to use a `View Object`.